### PR TITLE
[android]: nuke wifiManager

### LINKED ISSUE #5787: When you're not so close to WiFi ..
* KOReader version: 2019-12-27
* Device: Android (inkbook)

#### Issue

When you're not sitting next to wifi,  using online koreader functions can be very painfull:

https://youtu.be/mV6LyCLQtm8

often you have to click many times 
menu -> wallabag/newsdownloadr/send2ebook -> "retrievie articles from server" -> "turn on wifi" 
(and again, on example video this path was repeated around 7-8 , each 4 clicks, as above, gives 32 clicks) unit it really captures internet, and starts processing. That's horrible!

Year or 2 back I remember I did at least for newsDownloader or send2ebook callback function for enabling wifi that waited until it's connected (or terminates after 30sek) and then automaticly starts processing eg wallabag articles. 

But it seems it's not there anymore or is broken ? 
how about restoring that function and implementing it every where were you need to use "enable wifi" function ?

#### Steps to reproduce
described in issue

### PR DESCRIPTION
**Pull request by pazos**
Fixes #5787

---
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/5801)

### CODE REVIEW

File: frontend/device/android/device.lua [hunk 1]
```diff
@@ -254,12 +250,19 @@ function Device:performHapticFeedback(type)
...
-    if ip == "0" or gw == "0" then
+    local ok, type = android.getNetworkInfo()
+    if ok == "0" then
         return _("Not connected")
     else
-        local BD = require("ui/bidi")
-        return T(_("Connected to %1\n IP address: %2\n gateway: %3"), BD.wrap(ssid), BD.ltr(ip), BD.ltr(gw))
+        local status = nil
+        if type == "1" then
+            status = _("connected to Wi-Fi")
```
Comment (Frenzie) [inline frontend/device/android/device.lua]: Unless you want to change all the strings, it seems better to keep them consistent? :-)
```suggestion
            status = _("Connected to Wi-Fi")
```

File: frontend/device/android/device.lua [hunk 2]
```diff
@@ -254,12 +250,19 @@ function Device:performHapticFeedback(type)
...
+        elseif type == "2" then
+            status = _("connected to mobile data")
```
Comment (Frenzie) [inline frontend/device/android/device.lua]: Mobile data seems a bit odd? But we/I should double check what's used in Android and such too.
```suggestion
            status = _("Connected to mobile data network")
```

File: frontend/device/android/device.lua [hunk 3]
```diff
@@ -254,12 +250,19 @@ function Device:performHapticFeedback(type)
...
+        elseif type == "3" then
+            status = _("connected to ethernet")
+        end
+        return status or _("connecting")
```
Comment (Frenzie) [inline frontend/device/android/device.lua]: No point adding another string to translate.
```suggestion
        return status or _("Connecting…")
```

File: frontend/ui/network/manager.lua [hunk 4]
```diff
@@ -165,6 +165,7 @@ function NetworkMgr:setHTTPProxy(proxy)
 end
 
 function NetworkMgr:getWifiMenuTable()
+    if not Device:isAndroid() then
```
Comment (Frenzie) [inline frontend/ui/network/manager.lua]: Is this weird indentation here intentional? (If so I'd like to hear the rationale before saying it doesn't align with the coding style. :-P)

File: frontend/device/android/device.lua (hunk 2 above)
Comment (pazos) [reply frontend/device/android/device.lua]: I copied it from https://developer.android.com/training/monitoring-device-state/connectivity-status-type#DetermineType.

I also forgot WiMAX as a valid network type :p.

File: frontend/ui/network/manager.lua (hunk 4 above)
Comment (pazos) [reply frontend/ui/network/manager.lua]: intentional: lazyness porting from other branch. Will fix!

Comment (pazos) [reply frontend/ui/network/manager.lua]: Fixed!

File: frontend/device/android/device.lua [hunk 5]
```diff
@@ -230,22 +230,21 @@ end
 
 function Device:initNetworkManager(NetworkMgr)
     function NetworkMgr:turnOnWifi(complete_callback)
```
Comment (Frenzie) [inline frontend/device/android/device.lua]: It seems that if the `complete_callback` thing is useless here you could just as well write `NetworkMgr.turnOnWifi = android.openWifiSettings`? But whichever, writing it this way is probably friendlier to potential future changes.

File: frontend/device/android/device.lua [hunk 6]
```diff
@@ -254,12 +253,23 @@ function Device:performHapticFeedback(type)
...
-        local BD = require("ui/bidi")
-        return T(_("Connected to %1\n IP address: %2\n gateway: %3"), BD.wrap(ssid), BD.ltr(ip), BD.ltr(gw))
+        if type == C.ANETWORK_WIFI then
+            return _("Connected to Wi-Fi")
+        elseif type == C.ANETWORK_MOBILE then
+            return _("Connected to mobile data network")
+        elseif type == C.ANETWORK_ETHERNET then
+            return _("Connected to ethernet")
+        elseif type == C.ANETWORK_BLUETOOTH then
+            return _("Connected to bluetooth")
```
Comment (Frenzie) [inline frontend/device/android/device.lua]: As you know I'm in favor of wifi (and therefore possibly bluetooth by similar reasoning) but for now that should be Bluetooth in English.

```suggestion
            return _("Connected to Bluetooth")
```

File: frontend/device/android/device.lua (hunk 5 above)
Comment (pazos) [reply frontend/device/android/device.lua]: yup, ignoring is a valid way of treating a parameter :smile:

File: frontend/device/android/device.lua (hunk 6 above)
Comment (pazos) [reply frontend/device/android/device.lua]: The same for Ethernet?

**Comment by Frenzie** [comment]
https://github.com/koreader/koreader/pull/5801#discussion_r376160439
> The same for Ethernet?

Good point, if wifi is spelled Wi-Fi then ethernet should be spelled Ethernet.

On a related note, I just noticed this Microsoft styleguide: https://docs.microsoft.com/en-us/style-guide (naturally it's very trademarkophilic)

File: frontend/device/android/device.lua [hunk 7]
```diff
@@ -254,12 +253,23 @@ function Device:performHapticFeedback(type)
...
         return _("Not connected")
     else
-        local BD = require("ui/bidi")
-        return T(_("Connected to %1\n IP address: %2\n gateway: %3"), BD.wrap(ssid), BD.ltr(ip), BD.ltr(gw))
+        if type == C.ANETWORK_WIFI then
+            return _("Connected to Wi-Fi")
+        elseif type == C.ANETWORK_MOBILE then
+            return _("Connected to mobile data network")
+        elseif type == C.ANETWORK_ETHERNET then
+            return _("Connected to ethernet")
```
Comment (Frenzie) [inline frontend/device/android/device.lua]:
```suggestion
            return _("Connected to Ethernet")
```

**Pull request merged**